### PR TITLE
misc improvements to C jobspec library

### PR DESCRIPTION
**Pull request by garlick**
I started out tweaking the `flux_jobspec1` class added in #3662 to make sure the recent dependency definitions in RFC 26 were checked, but then went on to add some new functions to improve testing:
```c
/* Check the validity of the full jobspec, which may be necessary if the
 * jobspec object was created by flux_jobspec1_decode().
 * Return 0 on success, -1 on error with an error message written to errbuf.
 */
int flux_jobspec1_check (flux_jobspec1_t *jobspec, char *errbuf, int errbufsz);

/* Decode a string to jobspec.
 */
flux_jobspec1_t *flux_jobspec1_decode (const char *s, char *errbuf, size_t errbufsz);
```

A sharness test then pushes the jobspec test inputs through a little C program that runs decode/check.

Some other issues were addressed along the way.

### CODE REVIEW

**Review by chu11 — COMMENTED** [review]
took a quick skim and things LGTM, nothing stood out.  But perhaps @jameshcorbett should take a look to approve.

**Comment by jameshcorbett** [comment]
Looks great to me, thanks @garlick for cleaning things up!

**Review by chu11 — APPROVED** [review]

File: src/common/libjob/jobspec1.c [hunk 1]
```diff
@@ -78,7 +80,9 @@ int flux_jobspec1_attr_pack (flux_jobspec1_t *jobspec,
     return rc;
 }
 
-int flux_jobspec1_attr_check (flux_jobspec1_t *jobspec, char *errbuf, int errbufsz)
+int flux_jobspec1_attr_check (flux_jobspec1_t *jobspec,
+                              char *errbuf,
+                              int errbufsz)
```
Comment (grondo) [inline src/common/libjob/jobspec1.c:85]: I prefer the `*_error_t` style of error message passing over the `char *errbuf, int errbufsz` style. Maybe over time we could semi-standardize on that interface? IMO, removing the extra parameter makes it easy on the caller (among other benefits).

I realize this is just a style preference though.

Comment (garlick) [reply src/common/libjob/jobspec1.c:85]: I prefer that too.  OK, I'll see if I can make that change.

Comment (garlick) [reply src/common/libjob/jobspec1.c]: OK, just pushed a commit that makes that change.

**Comment by grondo** [comment]
Thanks @garlick!

typo in ea178c2: "alow" :arrow_right: "allow"

**Review by grondo — APPROVED** [review]
LGTM, Thanks!

**Comment by garlick** [comment]
Just pushed a fix for the commit typo (thanks).  I'll set MWP.

**Comment by garlick** [comment]
Restarted CI - one builder timed out downloading flux-security from github.

**Pull request merged**